src/main/java/org/truffleruby/interop/InteropNodes.java

```diff
@@ -939,11 +939,25 @@ public boolean internalBit(int bits) {
         @Specialization(guards = "existing")
         public int keyInfoFlagsToBitsNode(boolean existing, boolean readable, boolean writable,
                                           boolean invocable, boolean internal) {
-            return KeyInfo.newBuilder()
-                    .setReadable(readable)
-                    .setWritable(writable)
-                    .setInvocable(invocable)
-                    .setInternal(internal).build();
+            int keyInfo = 0;
+
+            if (readable) {
+                keyInfo |= KeyInfo.READABLE;
+            }
+
+            if (writable) {
+                keyInfo |= KeyInfo.MODIFIABLE;
+            }
+
+            if (invocable) {
+                keyInfo |= KeyInfo.INVOCABLE;
```

Comment:
Just a casual observation, but I was surprised to see this wasn't name `INVOKABLE`.